clamav/cnudie.py

```diff
@@ -14,29 +14,122 @@
 import clamav.scan
 import cnudie.iter
 import dso.model
+import delivery.model
+import delivery.client
 import github.compliance.model
 import oci.client
 
 logger = logging.getLogger(__name__)
 ci.log.configure_default_logging()
 
 
+def _finding_for_component_artifact(
```

Comment:
I have multiple concerns here:

- this function mixes two aspects: 
  - filter elements
  - return first element iff filtering result is not empty, or None
- this function is actually more generic (filters/finds artefact-metadata) than its location suggests (clamav/cnudie is specific to clamav)

Also, I think we should not hardcode the special-case of only considering the (randomly determined) first element (there could be multiple metadata-entries (in case of clamav scans, this could be very relevant, because newer results might lead to decision to skip, whereas older entries might not)).

--

Also, I would prefer to make use of `cnudie.iter`'s ResourceNode, ArtefactNode, ComponentNode for new code. As I commented below, it might be a good idea to add some convenience functions, e.g. to convert to and from cnudie.iter, dso.model, dm.model, name-and-version-strings (component_name, artefact_name, ....)

I think what you should factor out is a function that matches a cnudie.iter.XX-Node against dm.ArtefactMetadata

the specialcase-function in this module should instrument such matchers, to do special-case-filtering (if necessary), specific to clamav-metadata (e.g. based on timestamp, ..)